cpp/src/arrow/compute/kernels/take-test.cc

```diff
@@ -147,28 +181,129 @@ class TestTakeKernelWithString : public TestTakeKernel<StringType> {
     ASSERT_OK(DictionaryArray::FromArrays(type, ArrayFromJSON(int8(), expected_indices),
                                           dict, &expected));
     auto take_indices = ArrayFromJSON(int8(), indices);
-    this->AssertTakeArrays(values, take_indices, options, expected);
+    this->AssertTakeArrays(values, take_indices, expected);
   }
 };
 
 TEST_F(TestTakeKernelWithString, TakeString) {
-  TakeOptions options;
-  this->AssertTake(R"(["a", "b", "c"])", "[0, 1, 0]", options, R"(["a", "b", "a"])");
-  this->AssertTake(R"([null, "b", "c"])", "[0, 1, 0]", options, "[null, \"b\", null]");
-  this->AssertTake(R"(["a", "b", "c"])", "[null, 1, 0]", options, R"([null, "b", "a"])");
+  this->AssertTake(R"(["a", "b", "c"])", "[0, 1, 0]", R"(["a", "b", "a"])");
+  this->AssertTake(R"([null, "b", "c"])", "[0, 1, 0]", "[null, \"b\", null]");
+  this->AssertTake(R"(["a", "b", "c"])", "[null, 1, 0]", R"([null, "b", "a"])");
 
   std::shared_ptr<Array> arr;
-  ASSERT_RAISES(IndexError, this->Take(utf8(), R"(["a", "b", "c"])", int8(), "[0, 9, 0]",
-                                       options, &arr));
+  ASSERT_RAISES(IndexError,
+                this->Take(utf8(), R"(["a", "b", "c"])", int8(), "[0, 9, 0]", &arr));
+  ASSERT_RAISES(IndexError, this->Take(utf8(), R"(["a", "b", null, "ddd", "ee"])",
+                                       int64(), "[2, 5]", &arr));
```

Comment:
Do you sometimes test that a negative index raises `IndexError`?

Reply:
I'll add a test for that